File: app/rotas/sala.py
from app.rotas import *
from app.services.sala import *
from app.services.participante import *

def registerSala(app):
    
    # rota de primeiro acesso quando usuario entra pelo /
    @app.route('/sala/acessar/<token>', methods=['POST','GET'])
    def acessarSala(token):
        token = token.upper()
        resp = fentrarSala(token)
        return jsonify(resp)
    
    # rota alternativa de primeiro acesso quando o usuario entra pela leitura do qrcode ou link compartilhado
    @app.route('/sala/acess/<token>', methods=['POST','GET'])
    def acessSala(token):
        token = token.upper()

        resp = fentrarSala(token)

        if resp['msg'] == 'ok':
            return render_template('acessoTokenInput.html',qrcode=True, token=token)
        
        else:
            return render_template('paginaNaoEncontrada.html')
    
    @app.route('/sala/remover', methods=['DELETE'])
    def removerSala():
        respo = request.get_json()

        tokenSala = respo.get('tokenSala')

        data = fdeletarSala(tokenSala)

        return jsonify(data)
    
    @app.route('/sala/criar', methods=['POST'])
    def criarSala():
        data = request.get_json('data')

        data = data['data']

        respo = fcriarSala(data)

        return jsonify(respo)
    
    @app.route('/sala/carregar', methods=['GET'])
    def carregarSala():

        idUsuario = request.args.get('idUsuario')

        params = request.args.get('params')

        data = retornarSalasCriadas(idUsuario,params)

        return jsonify(data)
    
    @app.route('/sala/entrar', methods=['POST'])
    def entrarSala():
        data = request.get_json('data')
        nomeParticipante = data['nome']
        tokenSala = data['token']

        respo = participanteEntrarSala(nomeParticipante,tokenSala)

        return jsonify(respo)
    
    @app.route('/sala/visualizar/<token>', methods=['GET'])
    def visualizarSala(token):
        try:
            data = consultarSalaTokenDB(token)

            return render_template('painelSala.html',data=data)
        
        except Exception as e:
            print(e)

File: app/static/js/routes/sala.js
// requisição para acessar a sala primeiro passo
export async function acessarSala(token){
    // requisição para acessar a sala
    var response = await axios.post(`/sala/acessar/${token}`)

    // obtendo resposta do servidor
    var data = response.data

    return data
}

// requisição para entrar de fato na sala
export async function entrarSala(nome,token) {
    
    var response = await axios.post('/sala/entrar',{nome:nome, token:token})

    var data = response.data

    return data
}

// requisição para criar sala de oração
export async function criarSala(data) {
    
    var response = await axios.post('/sala/criar',{data:data})

    var data = response.data

    return data
}

// requisição para carregar todas as salas pertencentes ao usuario

export async function carregarSalas(idUsuario,params=null) {

    var response = await axios.get(`/sala/carregar`,{params:{idUsuario:idUsuario,params:params}})

    var data = response.data

    return data
}

// requisição para "deletar" sala de oração
export async function deletarSala(tokenSala) {
    
    var response = await axios.delete('/sala/remover', {data:{tokenSala:tokenSala}})

    var data = response.data

    return data
}

// requisição para visulizar painel da sala de oração
export async function visualizarSala(tokenSala) {

    var response = await axios.get('/sala/visualizar', {params:{tokenSala:tokenSala}})

    var data = response.data

    return data
    
}